src/core/validations/message.rs

```diff
@@ -34,6 +48,67 @@ pub fn validate_message(message: &proto::Message) -> Result<(), ValidationError>
         &message.signer,
     )?;
 
+    let network = FarcasterNetwork::try_from(message_data.network)
+        .or_else(|_| Err(ValidationError::InvalidData))?;
+
+    match &message_data.body {
+        Some(proto::message_data::Body::UserDataBody(_)) => {
+            // Split user data validation from engine
```

Comment:
This should have all the length validations from https://github.com/farcasterxyz/hub-monorepo/blob/913c0f67665e31241e45400b6c4f7471f28f75e2/packages/core/src/validations.ts#L977